### ISSUE TITLE
[Android][Quest2] Vulkan validation error: VUID-VkWriteDescriptorSet-descriptorType-00333

### ISSUE THREAD

**Issue by moraaar** [o3de/o3de#9398]
**Describe the bug**
When running an O3DE level on Oculus Quest 2 with Vulkan validation layer enabled the following error appears:

_2022-05-01 03:28:41.988 5919-5988/org.o3de.MinProject I/: <03:28:41> [Error] (vkDebugMessage) - ERROR: [Validation] Code 0 : Validation Error: [ VUID-VkWriteDescriptorSet-descriptorType-00333 ] Object 0: handle = 0x3811720000000258, type = VK_OBJECT_TYPE_DESCRIPTOR_SET; | MessageID = 0xf2fc081c | vkUpdateDescriptorSets() pDescriptorWrites[0] failed write update validation for VkDescriptorSet 0x3811720000000258[] with error: Write update to VkDescriptorSet 0x3811720000000258[] allocated with VkDescriptorSetLayout 0xbe87ec0000000256[] binding #0 failed with error message: Attempted write update to buffer descriptor failed due to: For buffer VkBuffer 0x96e20d000000028d[] VkDescriptorBufferInfo range is 268435456 which is greater than this device's maxStorageBufferRange (134217728). The Vulkan spec states: If descriptorType is VK_DESCRIPTOR_TYPE_STORAGE_BUFFER or VK_DESCRIPTOR_TYPE_STORAGE_BUFFER_DYNAMIC, the range member of each element of pBufferInfo, or the effective range if range is VK_WHOLE_SIZE, must be less than or equal to VkPhysicalDeviceLimits::maxStorageBuf_

**Steps to reproduce**
1. Build O3DE for Android
2. Add Vulkan-ValidationLayer binaries to the apk and enable it.
    i. Download Vulkan-ValidationLayer binaries from https://github.com/KhronosGroup/Vulkan-ValidationLayers/releases 
    ii. Copy the folder "arm64-v8a" into "LY_BUILD_DIR\app\src\main\jniLibs\"
    iii. Build again.
    iv. Validation layer will be enabled by default on debug configuration. To enable it on profile pass the following argument by command line "rhi-device-validation enable".
3. Deploy and run a level to Oculus Quest 2.

**Expected behavior**
No vulkan errors

**Actual behavior**
Vulkan errors printed in the log.

**Desktop/Device (please complete the following information):**
 - Device: Oculus Quest 2
 - OS: Android

**Comment by jeremyong-az** [comment]
Reading the error, apparently the maximum storage buffer size is 128MB and we're allocating 256MB somewhere (skinning data maybe)?

**Comment by moudgils** [comment]
yep. For Skinning we pre-allocate 256 MB. I think this needs to be dynamic. So if a scene does not have skinned meshes we shouldn't be allocating this buffer. I will bring this up internally.

**Comment by jeremyong-az** [comment]
At the RHI level, we probably need to cap allocations for storage buffers based on the advertised device requirements also. Should error/assert if this is requested.

**Comment by moudgils** [comment]
I will create a Jira for capping allocations for storage buffers based on the advertised device requirements

**Comment by moraaar** [comment]
I debugged this and the buffer size it's complaining about is controlled by the CVAR `r_skinnedMeshInstanceMemoryPoolSize` in `SkinnedMeshOutputStreamManager.cpp`. I tested to change the default value from 256MB to 128MB and that fixed the issue.

After discussing with @moudgils and @amzn-tommy we are going to address this in 2 steps:

1. To fix the issue now, we're going to PAL-ify the max buffer size per platform (256MB on PC/Linux/Max and 128MB on iOS/Android) so it can be checked, warned and caped the buffers allocated by `SkinnedMeshOutputStreamManager.cpp`

2. In the future replace the PAL-ification with a more dynamic approach. For example, adding API to RPI/RHI to request max buffer size, that way the upper level code ( SkinnedMeshOutputStreamManager.cpp ) can ask and cap the memory it requests for buffers dynamically.
